main.c

```diff
@@ -118,26 +148,27 @@ static const SerialConfig usart1Config = {
 
 static uint8_t commandBuffer[64];
 
-/*
- * Thread 1.
- */
-THD_WORKING_AREA(waThread1, 128);
-__attribute__((noreturn)) THD_FUNCTION(Thread1, arg) {
-  (void)arg;
+void updateLightningTimer(void) {
+    uint8_t freq = profiles[currentProfile].animationSpeed[currentSpeed];
+    if (freq > 0) {
+      if (GPTD_BFTM1.state == GPT_CONTINUOUS) {
+        if (gptGetIntervalX(&GPTD_BFTM1) != freq) {
+          gptStopTimer(&GPTD_BFTM1);
+          gptStartContinuous(&GPTD_BFTM1, freq);
```

Comment:
Technically I could use `gptChangeInterval(&GPTD_BFTM1, freq)` but based on my local testing it seems to be unstable.
The animation get stuck for some reason when I change speed/profile very quickly. With current implementation it seems to be stable.